File: golden_ticket_enterprise/lib/entities/group_member.dart

import 'package:golden_ticket_enterprise/entities/user.dart';

class GroupMember{
  final User? member;
  final DateTime? joinedAt;
  DateTime? lastSeenAt;
  GroupMember({required this.member, required this.joinedAt, required this.lastSeenAt});
  factory GroupMember.fromJson(Map<String, dynamic> json) {
    return GroupMember(
        member: User.fromJson(json['user']),
        joinedAt: DateTime.parse(json['joinedAt']),
        lastSeenAt: json['lastSeenAt'] != null ? DateTime.parse(json['lastSeenAt']) : null
    );
  }
}

File: GoldenTicket/GoldenTicket/Entities/GroupMember.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace GoldenTicket.Entities
{

    [Table("tblMember")]
    public class GroupMember
    {
        [Key] 
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ChatMemberID { get; set; }

        [Required]
        public int ChatroomID { get; set; }
        [ForeignKey("ChatroomID")]
        public Chatroom Channel { get; set; } = null!;

        [Required]
        public int MemberID { get; set; }

        [ForeignKey("MemberID")]
        public User? Member { get; set; }
        public DateTime JoinedAt {get;set;} = DateTime.Now;
        public DateTime? LastSeenAt { get; set; }
    }

    public class GroupMemberDTO { 
        public UserDTO User { get; set; }
        public DateTime? JoinedAt { get; set; }
        public DateTime? LastSeenAt { get; set; }
        public GroupMemberDTO(GroupMember groupMember){
            this.User = new UserDTO(groupMember.Member!);
            this.JoinedAt = groupMember.JoinedAt;
            this.LastSeenAt = groupMember.LastSeenAt ?? null;
        }
    }
}